src/syntax/ppx_sedlex.ml

```diff
@@ -23,6 +23,17 @@ type decision_tree =
   | Table of int * int array
   | Return of int
 
+let rec simplify_decision_tree ( x : decision_tree) = 
```

Comment:
Is this function specific to the code below? Should it remain inside the code's scope in this case or can it have other uses outside of it, in which can could its name be more explicit about what it's actually doing?

Reply:
it is used to avoid generating silly code like this ` if pred then 0 else 0`